components/articles/PublicArticle.js

```diff
@@ -2,61 +2,184 @@ import React from 'react';
 import renderHTML from 'react-render-html';
 import { connect } from 'react-redux';
 
+import Icon from 'components/common/Icon';
 import Link from 'components/common/Link';
+import Text from 'components/common/Text';
 
 import { selectors } from 'redux/ducks/articles';
 import { selectors as authSelectors } from 'redux/ducks/auth';
 import { ROUTES_NAMES } from 'routes';
 import { LOCALES } from 'constants';
+import { EXPORT_TO_NETWORS } from 'constants/social';
 import { ArticleModel } from 'utils/customPropTypes';
 
+import AuthorBlock from './public/AuthorBlock';
+import CollectionBlock from './public/CollectionBlock';
+import PublishInfo from './public/PublishInfo';
+import ShareToButton from './public/ShareToButton';
+
 const mapStateToProps = (state, { articleLocale }) => ({
-  canEditArticle: authSelectors.getPermissions(state).canManageArticles, // ??,
   otherLocales: selectors.getOtherLocales(state, articleLocale),
+  // TODO(andemerie): check if the following permission works as required
+  canEditArticle: authSelectors.getPermissions(state).canManageArticles,
 });
 
+// TODO(andemerie): move styles from markup repo to this repo and finish styles for this page
 const PublicArticle = ({
-  articleId,
+  collection,
+  canEditArticle,
   title,
-  subtitle,
-  imageUrl,
-  type,
-  brand,
-  text,
-  author,
+  articleId,
   articleLocale,
-  canEditArticle,
+  author,
+  // brand,
   otherLocales,
+  imageUrl,
+  text,
+  publishAt,
+  // TODO(andemerie): decide how to implement video page and where to check the following type
+  // type,
 }) => (
-  <div>
-    <h1>
-      {canEditArticle && (
-        <Link
-          route={ROUTES_NAMES.article}
-          params={{ slug: articleId, mode: 'edit', articleLocale }}
-        >
-          <a>Edit Article</a>
-        </Link>
-      )}
-    </h1>
-    <h4 className="text is-size-4 has-text-danger">TODO replace stub with real article page</h4>
-    <h2 className="text is-size-2 has-text-primary">{title}</h2>
-    <h3 className="text is-size-3">{subtitle}</h3>
-    <p>
-      чытайце таксама на:{' '}
-      {otherLocales.map(({ locale, slug, title: t }) => [
-        <Link key={locale} route={ROUTES_NAMES.article} params={{ slug }}>
-          <a title={t}>{LOCALES[locale]}</a>
-        </Link>,
-        ', ',
-      ])}
-    </p>
-    <p>
-      {type} article, {brand.name}, {author && `by ${author.displayName}`}
-    </p>
-    <img alt={title} src={imageUrl} />
-    <br />
-    {renderHTML(text)}
+  <div className="article-container container">
```

Comment:
such a big piece of code... sometimes I divide code by render functions... or, perhaps, it's possible to move something to separated component?

Reply:
@andemerie had it all in one file & I already extract 6 small components, could you suggest what else can be extracted?

Reply:
maybe build this code using functions?

Reply:
I hate this approach, cause you flat render separated by pieces and you should scroll file up & down every time

Reply:
Hm, okay, what about lines 
 - https://github.com/babajka/babajka-frontend/pull/26/files/3a1f5d97859a6f6b141b689b6978b426ad8a067e#diff-79633f853f85d9351784491cfcc6375aR52
- https://github.com/babajka/babajka-frontend/pull/26/files/3a1f5d97859a6f6b141b689b6978b426ad8a067e#diff-79633f853f85d9351784491cfcc6375aR142

we could create a component or render function (just one render function), this pieces of code have in common a lot

your point is reasonable, sometimes it really annoys
but, I believe, in this case it should be ok